cypress/e2e/opening-hours-editor.cy.ts

```diff
@@ -0,0 +1,298 @@
+  cy.get('button[title="Næste"]').click();
+};
+
+const navigateToMonthViewAdmin = () => {
+  cy.get(".fc-dayGridMonth-button").click();
+};
+
+const selectTodayFromMonthViewAdmin = () => {
+  cy.get(".fc-day-today").click();
+};
+
+const selectTimeOnThursdayFromWeekView = ({
+  start,
+}: Pick<OpeningHourFormType, "start">): void => {
+  // In FullCalendar, the date and time elements are siblings in the same overlaying div, which prevents selection by both date and time simultaneously.
+  // To work around this, we target a specific time slot. This example selects the 08:00 slot, which spans all days.
+  // Since Cypress clicks at the center of the target element by default, and our time slots extend across all weekdays, it will interact with the slot for Thursday.
+  cy.get(`td.fc-timegrid-slot-lane[data-time="${start}:00"]`).click();
+};
+
+const fillOpeningHourForm = ({
+  openingHourCategory,
+  start,
+  end,
+}: Partial<OpeningHourFormType>) => {
+  cy.getBySel("opening-hours-editor-form").should("be.visible");
+
+  if (openingHourCategory) {
+    cy.getBySel("opening-hours-editor-form-select").select(openingHourCategory);
+  }
+  if (start) {
+    cy.getBySel("opening-hours-editor-form-start-time").focus().type(start);
+  }
+  if (end) {
+    cy.getBySel("opening-hours-editor-form-end-time").focus().type(end);
+  }
+};
+
+const submitOpeningHourForm = () => {
+  cy.getBySel("opening-hours-editor-form-submit").click();
+};
+
+const checkOpeningHoursAdmin = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  return cy
+    .getBySel("opening-hours-editor-event-content")
+    .should("be.visible")
+    .and("contain", openingHourCategory)
+    .and("contain", `${start} - ${end}`);
+};
+
+const checkOpeningHoursPage = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  cy.getBySel("opening-hours-week-list")
+    .should("be.visible")
+    .and("contain", openingHourCategory)
+    .and("contain", `${start} - ${end}`);
+};
+
+const checkOpeningHoursNotPresentInPage = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  cy.getBySel("opening-hours-week-list")
+    .should("be.visible")
+    .should("not.contain", openingHourCategory)
+    .should("not.contain", `${start} - ${end}`)
+    .contains("The library is closed this day");
+};
+
+const createOpeningHour = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  visitOpeningHoursPageAdmin();
+  navigateToMonthViewAdmin();
+  selectTodayFromMonthViewAdmin();
+  fillOpeningHourForm({ openingHourCategory, start, end });
+  submitOpeningHourForm();
+  checkOpeningHoursAdmin({ openingHourCategory, start, end });
+  visitOpeningHoursPage();
+  checkOpeningHoursPage({ openingHourCategory, start, end });
+};
+
+const createOpeningHourInNextWeek = ({
+  openingHourCategory,
+  start,
+  end,
+}: OpeningHourFormType) => {
+  visitOpeningHoursPageAdmin();
+  navigateToNextWeekAdmin();
```

Comment:
It would be nice to pass a date for the opening hours to this function but I guess this becomes complex.